### PR TITLE
Implement dry run mode for upgrade

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Implement dry run mode for upgrade
- Fixed some issues in orchestration implemetation

You need to specify the `dry` parameter in the request body to `true` in the orchestration call to run upgrade in dry run mode 

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/orchestration/kyma/manager.go [hunk 1]
```diff
@@ -58,17 +58,21 @@ func (u *upgradeKymaManager) Execute(orchestrationID string) (time.Duration, err
...
+	switch {
+	case err != nil:
 		return 0, errors.Wrap(err, "while updating orchestration")
-	}
-	if repeat != 0 {
+	case repeat != 0:
 		return repeat, nil
-	}
-	if len(operations) == 0 {
+	case isFinished:
```
Comment (jasiu001) [inline components/kyma-environment-broker/internal/orchestration/kyma/manager.go:75]: if `dto.Dry` will be true then the strategy will not trigger the `Execute` method with all upgrade processes. How will we see what is the result of all upgrade process?

**Review by jasiu001 — CHANGES_REQUESTED** [review]

File: components/kyma-environment-broker/internal/orchestration/dto.go [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+import (
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+)
 
 type Parameters struct {
 	Targets  internal.TargetSpec   `json:"targets"`
 	Strategy internal.StrategySpec `json:"strategy,omitempty"`
+	Dry      bool                  `json:"dry,omitempty"`
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/orchestration/dto.go:12]: I would change the name to "DryRun"

File: components/kyma-environment-broker/internal/orchestration/handlers/converter_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package handlers_test
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/orchestration/handlers/converter_test.go:1]: minor: add "given", "when", "then"

**Review by jasiu001 — APPROVED** [review]

**Review by jasiu001 — APPROVED** [review]

**Pull request merged**